src/OnboardingSPA/components/Drawer/DrawerPanel/DesignColors.js

```diff
@@ -85,28 +238,20 @@ const DesignColors = () => {
 			selectedGlobalStyle?.settings?.color?.palette;
 		if ( colorPalettesTemp && colorStyle && selectedThemeColorPalette ) {
 			for ( let idx = 0; idx < selectedThemeColorPalette.length; idx++ ) {
+				// Iterate through every element in array
+				// i.e. [{color: '#ffffff', name: 'Base', slug: 'base'}, ...] and update the appropriate color that has been changed
 				const slug = selectedThemeColorPalette[ idx ]?.slug;
 				if (
-					isCustomStyle &&
-					selectedColorsLocalTemp?.[ slug ] !== '' &&
-					selectedColorsLocalTemp?.[ slug ] !== undefined
-				) {
-					selectedThemeColorPalette[ idx ].color =
-						selectedColorsLocalTemp[ slug ];
-				} else if (
-					// Add Exception for Background. (perhaps scope to yith-wonder in future)
-					colorPalettesTemp?.[ colorStyle ]?.[ slug ] &&
-					'base' === slug
-				) {
-					selectedThemeColorPalette[ idx ].color = '#FFFFFF';
-				} else if (
 					! isCustomStyle &&
```

Comment:
Why check for `isCustomStyle`? Shouldn't that happen before determining whether to call `saveThemeColorPalette` vs `saveCustomColors`?